pkgs/top-level/php-packages.nix

```diff
@@ -92,13 +94,27 @@ let self = with self; {
     buildInputs = [ pkgs.m4 ];
   };
 
-  pthreads = assert pkgs.config.php.zts or false; buildPecl {
-    #pthreads requires a build of PHP with ZTS (Zend Thread Safety) enabled
-    #--enable-maintainer-zts or --enable-zts on Windows
+  #pthreads requires a build of PHP with ZTS (Zend Thread Safety) enabled
+  #--enable-maintainer-zts or --enable-zts on Windows
+  pthreads = if php7 then pthreads31 else pthreads20;
+
+  # 2.x is not supported and does not work on PHP 7
+  pthreads20 = assert pkgs.config.php.zts && (!php7); buildPecl {
     name = "pthreads-2.0.10";
     sha256 = "1xlcb1b1g10jd0xhm3c01a06yqpb5qln47pd1k522138324qvpwb";
   };
 
+  # Starting from 3.x, pthreads requires at least PHP 7.0
+  pthreads30 = assert pkgs.config.php.zts && php7; buildPecl {
```

Comment:
Do we have users actually relying on this version? If not we should wait to add it until we need to.